docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install

```diff
@@ -25,308 +29,444 @@ function su_humsci_profile_final_task() {
+    }
+  }
 }
 
 /**
- * Disable preview mode on all content types.
+ * Migrate ECKs to the new types.
  */
-function su_humsci_profile_update_8203() {
-  $node_types = \Drupal::entityTypeManager()
-    ->getStorage('node_type')
-    ->loadMultiple();
-  $config_factory = \Drupal::configFactory();
-  foreach (array_keys($node_types) as $type) {
-    $config_factory->getEditable("node.type.$type")
-      ->set('preview_mode', 0)
-      ->save();
+function su_humsci_profile_update_9701(&$sandbox) {
+  $entity_type_manager = \Drupal::entityTypeManager();
+  $eck_info = _su_humsci_profile_update_eck_info();
+  if (!isset($sandbox['count'])) {
+    $sandbox['items'] = [];
+    $sandbox['convert'] = [];
+
+    // Store all the ECK ids into the sandbox for batch processing.
+    foreach (array_keys($eck_info) as $eck_name) {
+      $ids = $entity_type_manager->getStorage($eck_name)
+        ->getQuery()
+        ->accessCheck(FALSE)
+        ->execute();
+      foreach ($ids as $id) {
+        $sandbox['items'][] = "$eck_name:$id";
+      }
+    }
+    $sandbox['items'] = array_filter($sandbox['items']);
+    $sandbox['count'] = count($sandbox['items']);
+    \Drupal::messenger()
+      ->addStatus(t('Updating @count ECK items.', ['@count' => count($sandbox['items'])]));
   }
-}
 
-/**
- * Disable hs_revision_cleanup module.
- */
-function su_humsci_profile_update_8204() {
-  \Drupal::service('module_installer')->uninstall(['hs_revision_cleanup']);
-}
+  $batched_items = array_splice($sandbox['items'], 0, 100);
+  foreach ($batched_items as $item) {
+    [$eck_type, $id] = explode(':', $item);
+    $entity = $entity_type_manager->getStorage($eck_type)->load($id);
+
+    $new_bundle_id = $eck_info[$eck_type][$entity->bundle()];
+    /** @var \Drupal\hs_entities\HsEntityInterface $new_entity */
+    $new_entity = \Drupal::entityTypeManager()
+      ->getStorage('hs_entity')
+      ->create([
+        'label' => $entity->label(),
+        'bundle' => $new_bundle_id,
+      ]);
+
+    // Make sure not to copy the id or uuid fields from the ECK.
+    $skip_fields = ['id', 'uuid'];
+    // Copy any fields that exist on both entity types.
+    foreach (array_keys($entity->getFieldDefinitions()) as $field_name) {
+      if ($new_entity->hasField($field_name) && !in_array($field_name, $skip_fields)) {
+        $new_entity->set($field_name, $entity->get($field_name)->getValue());
+      }
+    }
+    $new_entity->save();
+    $sandbox['convert'][$eck_type][$id] = $new_entity->id();
+    $entity->delete();
+  }
+  $sandbox['#finished'] = empty($sandbox['items']) ? 1 : ($sandbox['count'] - count($sandbox['items'])) / $sandbox['count'];
 
-/**
- * Uninstall deprecated webform_jqueryui_buttons module.
- */
-function su_humsci_profile_update_8205() {
-  \Drupal::service('module_installer')->uninstall(['webform_jqueryui_buttons']);
-}
+  // When the conversion is complete, update all the field data.
+  if (!empty($sandbox['items'])) {
+    return;
+  }
```

Comment:
I assume the return here just restarts the batch process on the next batch, so that once all the batching completes it executes the entity reference id migration below.

Reply:
Yep that exactly what's happening here. We can't pass sandbox to the next update hook so we gotta stay in the same one.